infra/cron/src/jobs/revertAlertJob.js

```diff
@@ -0,0 +1,187 @@
+
+  logger.debug(`Fetching ${url}`)
+
+  const res = await fetch(url)
+
+  if (res.status !== 200) {
+    throw new Error('Unable to query etherscan')
+  }
+
+  const jason = await res.json()
+
+  if (!jason.status) {
+    throw new Error('Query to etherscan failed. Invalid resposne.')
+  }
+
+  return jason.result
+}
+
+/**
+ * Returns a date string in ISO8601
+ * @returns {string} today's date
+ */
+function todaysDateString() {
+  const now = new Date()
+  const year = now.getUTCFullYear()
+  const month = String(now.getUTCMonth()).padStart(2, '0')
+  const day = String(now.getUTCDate()).padStart(2, '0')
+  return `${year}-${month}-${day}`
+}
+
+function block24HoursAgo(blockNumber) {
+  return blockNumber - DAILY_BLOCKS
+}
+
+async function RevertAlertJob(job) {
+  logger.info(`Processing RevertAlertJob id=${job.id} data= ${job.data}`)
+  job.progress(0)
+
+  if (!['mainnet', 'rinkeby'].includes(net)) {
+    job.progress(100)
+    return { jobDone: false }
+  }
+
+  if (typeof process.env.ETHERSCAN_API_KEY === 'undefined') {
+    throw new Error('Env var ETHERSCAN_API_KEY must be set')
+  }
+  if (typeof process.env.EMAILS_TO === 'undefined') {
+    throw new Error('Warning: EMAILS_TO env var is not set')
+  }
+  if (typeof process.env.SENDGRID_API_KEY === 'undefined') {
+    throw new Error('Env var SENDGRID_API_KEY must be set')
+  }
+  if (typeof process.env.SENDGRID_FROM_EMAIL === 'undefined') {
+    logger.warn('Warning: SENDGRID_FROM_EMAIL env var is not set')
+  }
+
+  job.progress(5)
+
+  setNetwork(net)
+  if (!contractsContext.web3) {
+    throw new Error('web3 not initialized')
+  }
+
+  const latestBlock = await contractsContext.web3.eth.getBlockNumber()
+  const yesterdaysBlock = block24HoursAgo(latestBlock)
+
+  job.progress(10)
+
+  sendgridMail.setApiKey(process.env.SENDGRID_API_KEY)
+
+  const apiKey = process.env.ETHERSCAN_API_KEY
+
+  let total = 0
+  const reverts = []
+  const contracts = []
+  contracts.push({
+    address: contractsJson['Marketplace'],
+    epoch: contractsJson['MarketplaceEpoch']
+  })
+  if (contractsJson['Marketplace_V01']) {
+    contracts.push({
+      address: contractsJson['Marketplace_V01'],
+      epoch: contractsJson['MarketplaceEpoch_V01']
+    })
+  }
+
+  job.progress(30)
+
+  for (const c of contracts) {
+    const { address, epoch } = c
+    const fromBlock = Math.max(epoch, yesterdaysBlock)
+
+    const transactions = await fetchTransactions(address, {
+      fromBlock,
+      endBlock: latestBlock,
+      apiKey
+    })
+
+    for (const tx of transactions) {
+      total += 1
+      if (tx.txreceipt_status !== '1') {
+        reverts.push(tx.hash)
+      }
+    }
+
+    const internals = await fetchTransactions(c.address, {
+      action: 'txlistinternal',
+      fromBlock,
+      apiKey
+    })
+
+    for (const tx of internals) {
+      total += 1
+      if (tx.isError === '1') {
+        reverts.push(tx.hash)
+      }
+    }
+  }
+
+  // test revert
+  // reverts.push('0xdeadbeef')
```

Comment:
nit: remove

Reply:
eh, useful for testing, kind of left it intentionally for later testing.